client/extensions/woocommerce/state/sites/product-categories/selectors.js

```diff
@@ -1,19 +1,40 @@
 /**
  * External dependencies
  */
-import { get, isArray } from 'lodash';
+import { get, find, isArray } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
 
 /**
  * Gets product categories from API data.
  *
- * @param {Object} state Global state tree
- * @param {Number} siteId wpcom site id
+ * @param {Object} rootState Global state tree
+ * @param {Number} [ siteId ] wpcom site id, if not provided, uses the selected site id.
  * @return {Array} List of product categories
  */
-export function getProductCategories( state, siteId ) {
-	const categories = get( state, [ 'extensions', 'woocommerce', 'sites', siteId, 'productCategories' ], [] );
+export function getProductCategories( rootState, siteId = getSelectedSiteId( rootState ) ) {
+	const categories = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId, 'productCategories' ], [] );
 	if ( ! isArray( categories ) ) {
 		return [];
 	}
 	return categories;
 }
+
+/**
+ * Gets product category fetched from API data.
+ *
+ * @param {Object} rootState Global state tree
+ * @param {Number} categoryId The id of the category sought
+ * @param {Number} siteId wpcom site id
+ * @return {Object|null} Product category if found, otherwise null.
+ */
+export function getProductCategory( rootState, categoryId, siteId = getSelectedSiteId( rootState ) ) {
+	const categories = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId, 'productCategories' ], [] );
+	if ( ! isArray( categories ) ) {
+		return;
+	}
+	return find( categories, ( c ) => categoryId === c.id ) || null;
```

Comment:
Instead of the function you could do something simpler like

`find( categories, { 'id': categoryId } );`